src/megameklab/com/ui/Aero/tabs/EquipmentTab.java

```diff
@@ -176,15 +173,12 @@ public EquipmentTab(EntitySource eSource) {
         masterEquipmentList = new EquipmentTableModel(eSource.getEntity(), eSource.getTechManager());
         masterEquipmentTable.setModel(masterEquipmentList);
         masterEquipmentTable.setAutoResizeMode(JTable.AUTO_RESIZE_ALL_COLUMNS);
-        equipmentSorter = new TableRowSorter<EquipmentTableModel>(masterEquipmentList);
-        equipmentSorter.setComparator(EquipmentTableModel.COL_HEAT, new WeaponIntegerSorter());
-        equipmentSorter.setComparator(EquipmentTableModel.COL_MRANGE, new WeaponIntegerSorter());
-        equipmentSorter.setComparator(EquipmentTableModel.COL_DAMAGE, new WeaponDamageSorter());
-        equipmentSorter.setComparator(EquipmentTableModel.COL_RANGE, new WeaponRangeSorter());
-        equipmentSorter.setComparator(EquipmentTableModel.COL_COST, new FormattedNumberSorter());
-        equipmentSorter.setComparator(EquipmentTableModel.COL_TON, new FormattedNumberSorter());
+        equipmentSorter = new TableRowSorter<>(masterEquipmentList);
+        for (int col = 0; col < EquipmentTableModel.N_COL; col++) {
+            equipmentSorter.setComparator(col, masterEquipmentList.getSorter(col));
+        }
         masterEquipmentTable.setRowSorter(equipmentSorter);
-        ArrayList<RowSorter.SortKey> sortKeys = new ArrayList<RowSorter.SortKey>();
+        ArrayList<RowSorter.SortKey> sortKeys = new ArrayList<>();
```

Comment:
Can this be a List?

Reply:
Why? It's a local variable that is assigned a value once when it's instantiated.

Reply:
Best practices are to use the List interface for the variable declaration unless you need something specific to ArrayList?

Reply:
It makes a difference in method parameters, where ArrayList would prevent passing any List as an argument that is not instanceof ArrayList, or when the variable might be assigned the results of an List expression, but neither is the case here. If I were writing this as new code I would declare it as List as habit, but as there is no problem with the way it was written and there's no reason to change it.